include/arch/x86/arch.h

```diff
@@ -576,7 +577,7 @@ static inline u32_t _arch_syscall_invoke6(u32_t arg1, u32_t arg2, u32_t arg3,
 #endif
 			 : "S" (call_id), "a" (arg1), "d" (arg2),
 			   "c" (arg3), "b" (arg4), "D" (arg5),
-			   [arg6] "m" (arg6));
```

Comment:
You could also try making this an in-out argument ([arg6] "+m" (arg6)), maybe with early clobber ("+&m").

Using volatile looks hackish and fragile. 

Another suggestion is:

```c
    register u32_t a6 asm("ebp") = arg6;
    __asm__ volatile("int $0x80")"
 			 : "=a" (ret)
 #ifdef CONFIG_X86_IAMCU
 			   , "=d" (arg2), "=c" (arg3)
  #endif
  			 : "S" (call_id), "a" (arg1), "d" (arg2),
  			   "c" (arg3), "b" (arg4), "D" (arg5));
```

Reply:
> You could also try making this an in-out argument ([arg6] "+m" (arg6)), maybe with early clobber ("+&m").

I tried various combinations of these; none of them fix the issue

Reply:
> register u32_t a6 asm("ebp") = arg6;

This doesn't work either